src/na/na_ofi.c

```diff
@@ -508,28 +508,46 @@ na_ofi_av_insert(na_class_t *na_class, char *node_str, char *service_str,
     return ret;
 }
 
-/* lookup the address hash-table */
+static na_uint64_t
+key_mix(na_uint64_t key, na_uint64_t magic)
+{
+    /**
+     * http://www.partow.net/programming/hashfunctions/
+     */
+    return (magic * key) ^ ((magic << 3) + (key >> 2));
+}
+
+/* addr_str is in the format: "fi_addr_psmx2://ffff:ff02" */
+static na_uint64_t
+psm2_straddr_2_key(const char *addr_str)
```

Comment:
i have not went through other change, just see this one firstly.
The key has to be able to unique identify an endpoint, any kind of hash cannot fully ensure this.
you may refer some details in PSM2 address encoding to see if can parse/encode to get a unique 64bits key, if cannot, may need to use a 128 bits key.